app/services/store/builders.py

```
@@ -0,0 +1,403 @@
+view models per SPEC-UI-001 and INV-ARC-022.
+
+All business logic for rent entitlements, pricing, and collective goal progress
+is pre-computed here. Templates receive only formatted display values.
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from decimal import Decimal
+from datetime import datetime
+from typing import Any, Collection
```

Ruff (0.16.1)

[warning] Import from `collections.abc` instead: `Collection`

Import from `collections.abc`

(UP035)